### ISSUE TITLE
Lenient mode errors for returning nullable values from null-unmarked methods with a type-parameter return type

### ISSUE THREAD

**Issue by cpovirk** [jspecify/nullness-checker-for-checker-framework#172]
At the current main-eisop (76b33c9872c1ccc1b0fc9219ed70b2237b4e6db3) and still with #171:

```java
class Issue172<E> {
  E e() {
    return null;
  }
}
```

```
$ ./demo Issue172.java
Issue172.java:3: error: [return.type.incompatible] incompatible types in return.
    return null;
           ^
  type of expression: null?
  method return type: E
1 error
```

This should be allowed because the return type should be read as `E*` (i.e., `E` with unspecified nullness), which permits `null` in lenient mode.

Since this error comes up only in null-unmarked code, we should be OK with suppressing the small number of times that it occurs. Still, I'm filing it for tracking.

**Comment by cpovirk** [comment]
(https://github.com/jspecify/jspecify-reference-checker/commit/76b33c9872c1ccc1b0fc9219ed70b2237b4e6db3#diff-ce82a59b06d0d82bfc6d06835d1b06ac15619bc3a71617d32189d041f02ca9c0R921 came to mind because I remembered seeing `nullnessOperatorUnspecified`, but undoing that change does not help.)

**Comment by cpovirk** [comment]
I've realized that my earlier testing was incomplete. And it turns out that this problem comes up more when I test fully.

On the bright side, my testing with #175 (well, with [my variant of it](https://github.com/jspecify/jspecify-reference-checker/pull/175#discussion_r1567591777)) does appear to eliminate those errors. So we still look to be in good shape.

**Comment by cpovirk** [comment]
Yes, once #171 and #175 have both landed, I think we'll be ready to switch over.

**Comment by cpovirk** [comment]
(Or, come to think of it, I might be able to _mostly_ work around this one if necessary. But I'm not sure, and it would be at least a little work.)

**Comment by wmdietl** [comment]
Fixed by #175.
